pkg/prometheus/resource_selector.go

```diff
@@ -1044,7 +1044,37 @@ func (rs *ResourceSelector) validateDNSSDConfigs(sc *monitoringv1alpha1.ScrapeCo
 }
 
 func (rs *ResourceSelector) validateEC2SDConfigs(ctx context.Context, sc *monitoringv1alpha1.ScrapeConfig) error {
+	promVersion := operator.StringValOrDefault(rs.p.GetCommonPrometheusFields().Version, operator.DefaultPrometheusVersion)
+	version, err := semver.ParseTolerant(promVersion)
+	if err != nil {
+		return fmt.Errorf("failed to parse Prometheus version: %w", err)
+	}
+	if version.LT(semver.MustParse("0.16.0")) {
+		return fmt.Errorf("EC2 SD configuration is only supported for Prometheus version >= 0.16.0")
```

Comment:
in practice, we don't reject an object when it uses a field which isn't supported by the deployed Prometheus version. Instead we skip the field in the generated config.

Reply:
hmm... I think earlier in one of the other PRs (I can't recall which one exactly) I got the feedback that we need to reject the object if it contains fields that are not supported. Hence I tried to shift the version checks to `resource_selector.go` in the newer refactor PRs. 
Do you think we need to change it up?

Reply:
TLDR; If we are rejecting the objects which contains unsupported service discovery configurations (as mentioned [here](https://github.com/prometheus-operator/prometheus-operator/pull/6586#discussion_r1621961650)), I can not see why we should do the same with other unsupported fields.